pkg/datapath/linux/probes/probes.go

```diff
@@ -422,3 +395,101 @@ func (p *ProbeManager) KernelConfigAvailable() bool {
 
 	return true
 }
+
+// HaveMapType is a wrapper around features.HaveMapType() to check if a certain
+// BPF map type is supported by the kernel.
+// On unexpected probe results this function will terminate with log.Fatal().
+func HaveMapType(mt ebpf.MapType) error {
+	err := features.HaveMapType(mt)
+	if errors.Is(err, ebpf.ErrNotSupported) {
+		return err
+	}
+	if err != nil {
+		log.WithError(err).Fatalf("failed to probe MapType: %s", mt.String())
```

Comment:
I don't particularly like combining the check for the map type with the forced exit here. For instance, we would know that the map type is missing. But we don't know _why_ that map type is needed. Maybe there's a workaround that involves disabling some feature flag. Only the caller has that context. If we introduce a new map type tomorrow that can be conditionally used depending on which kernel Cilium runs on, then this also prevents us from having an implementation that conditionally checks for that support and falls back to an older / more naive implementation. This is the case today for LRU and LPM map types, for what it's worth. I anticipate we'll need something like this in future too.

I would suggest creating some canonical error here, so above declare a `ErrorMapTypeUnsupported = fmt.Errorf("map type unsupported")` and then here, do:

```suggestion
        return fmt.Errorf("failed to probe MapType %s: %w", mt.String(), ErrorMapTypeUnsupported)
```

Then, the calling code could do `if errors.Is(err, probes.ErrorMapTypeUnsupported) { ... fallback / exit ... }`.

Reply:
(I realize now that `ebpf` library already has the unsupported error so we probably don't need to redefine it)

Reply:
ditto, I just missed the `ErrNotSupported` logic. More :coffee: required.